src/Compilers/Core/Portable/DiagnosticAnalyzer/SuppressMessageAttributeState.cs

```diff
@@ -97,6 +95,25 @@ public static Diagnostic ApplySourceSuppressions(Diagnostic diagnostic, Compilat
             return diagnostic;
         }
 
+        public static bool IsDiagnosticSuppressed(Diagnostic diagnostic, Compilation compilation, out AttributeData suppressingAttribute)
+        {
+            SuppressMessageInfo info;
+            if (IsDiagnosticSuppressed(diagnostic, compilation, out info))
+            {
+                suppressingAttribute = info.Attribute;
+                return true;
+            }
+
+            suppressingAttribute = null;
+            return false;
+        }
+
+        private static bool IsDiagnosticSuppressed(Diagnostic diagnostic, Compilation compilation, out SuppressMessageInfo info)
```

Comment:
Yes probably, but keeping it as it is now as it is a private method, will cleanup all code in this file in a separate change.